File: src/test/java/com/lkimilhol/healingwords/words/domain/TextTest.java
package com.lkimilhol.healingwords.words.domain;

import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import com.lkimilhol.healingwords.exception.TextLengthExceedException;

class TextTest {

    @DisplayName("최대 길이 테스트")
    @Test
    void maxLength() {
        // given
        int index = 0;
        StringBuilder content = new StringBuilder();

        // when
        while (index++ < Text.MAX_LENGTH + 1) {
            content.append("t");
        }

        // then
        assertThatThrownBy(() -> new Text(content.toString()))
                .isInstanceOf(TextLengthExceedException.class);
    }
}

File: src/test/kotlin/words/domain/TextTest.kt
package words.domain

import exception.TextLengthExceedException
import io.kotlintest.shouldBe
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("글 작성 테스트")
internal class TextTest {

    @Test
    fun `작성 실패`() {
        // given
        var index = 0
        val content = StringBuilder()

        // when
        while (index++ < Text.MAX_LENGTH + 1) {
            content.append("t")
        }

        // then
        assertThatThrownBy { Text.create(content.toString()) }
            .isInstanceOf(TextLengthExceedException::class.java)
    }

    @Test
    fun `작성 성공`() {
        // given
        val content = "글작성"

        // when
        val text = Text.create(content)

        // then
        text.content() shouldBe content
    }
}